docs/feature_oled_driver.md

```diff
@@ -96,17 +96,19 @@ void oled_task_user(void) {
-| `OLED_FONT_END`        | `224`             | The ending characer index for custom fonts                                                                                 |
-| `OLED_FONT_WIDTH`      | `6`               | The font width                                                                                                             |
-| `OLED_FONT_HEIGHT`     | `8`               | The font height (untested)                                                                                                 |
-| `OLED_DISABLE_TIMEOUT` | *Not defined*     | Disables the built in OLED timeout feature. Useful when implementing custom timeout rules.                                 |
-| `OLED_IC`              | `OLED_IC_SSD1306` | Set to `OLED_IC_SH1106` if you're using the SH1106 OLED controller.                                                        |
-| `OLED_COLUMN_OFFSET`   | `0`               | (SH1106 only.) Shift output to the right this many pixels.<br />Useful for 128x64 displays centered on a 132x64 SH1106 IC. |
+| Define                  | Default           | Description                                                                                                                |
+|-------------------------|-------------------|----------------------------------------------------------------------------------------------------------------------------|
+| `OLED_DISPLAY_ADDRESS`  | `0x3C`            | The i2c address of the OLED Display                                                                                        |
+| `OLED_FONT_H`           | `"glcdfont.c"`    | The font code file to use for custom fonts                                                                                 |
+| `OLED_FONT_START`       | `0`               | The starting characer index for custom fonts                                                                               |
+| `OLED_FONT_END`         | `224`             | The ending characer index for custom fonts                                                                                 |
+| `OLED_FONT_WIDTH`       | `6`               | The font width                                                                                                             |
+| `OLED_FONT_HEIGHT`      | `8`               | The font height (untested)                                                                                                 |
+| `OLED_TIMEOUT`          | `60000`           | Turns off the OLED screen after 60000ms of keyboard inactivity. Helps reduce OLED Burn-in. Set to 0 to disable.            |
+| `OLED_SCROLL_TIMEOUT`   | `0`               | Scrolls the OLED screen after 0ms of OLED inactivity. Helps reduce OLED Burn-in. Set to 0 to disable.                      |
+| `OLED_SCROLL_TIMEOUT_FN`| `oled_scroll_left`| Scroll direction when `OLED_SCROLL_TIMEOUT` is triggered.                                                                  |
```

Comment:
Instead of a function pointer, would this not be more consistent as an explicit LEFT/RIGHT and a weak function that can be overridden for any potential custom behaviour?

Reply:
Well, it's not used as a function pointer, just a way to inline the specific function to call, but I see where you are going with this. An explicit LEFT/RIGHT define would be a safer option as then a user couldn't just inject whatever in that define. 

Now using a weak function I feel like is an unsafe option or as this timeout is directly linked to the internal state of the oled driver. The code that triggers the scroll on timeout uses the internal `oled_scrolling` field to track if the timeout has triggered or not to stop the timer from trying to continually checking. So without exposing internal state to everyone, you can easily run into cases where the function was overridden but does not start scrolling, then we end up in a case where it would trigger almost indefinitely.

```
#if OLED_SCROLL_TIMEOUT > 0
  if (!oled_scrolling && timer_expired32(timer_read32(), oled_scroll_timeout)) {
    OLED_SCROLL_TIMEOUT_FN();
  }
#endif
```

Reply:
Updated the define to not be a function name